### DISCUSSION TITLE
profiles and compile/build

### DISCUSSION THREAD

**Discussion by M0rr155** [vscode-abl/vscode-abl#145] [Q&A] [answered]
Our project exists of gui and tty parts.
i have created 2 profiles:

    "profiles": [
      { "name": "V12.8 GUI", "value": { "oeversion": "12.8", "graphicalMode": true }},
      { "name": "V12.8 TTY", "value": { "oeversion": "12.8", "graphicalMode": false }}
    ]

Now how can i get the build to build a specific part with tty settings/profile?

    "buildPath": [
      // Entries can have type 'source' or 'propath'. Path attribute is mandatory. Build attribute is optional (defaults to 'path'). Xref attribute is optional (defaults to 'build/.pct' or '.builder/srcX')
      { "type": "source" , "path": "srcNG", "excludes": "UpdProcs/ARCHIVE/**,UpdProcs/R200/**", "build": "build"},
      { "type": "source" , "path": "src/gen", "build": "build/gen"},
      { "type": "source" , "path": "src/gui", "build": "build/gui"},

PS: how can i turn of  Xref attribute?

**Comment by gquerret** [comment, marked as answer]
buildPath can be overridden in profiles:
```
"profiles": [
  { "name": "V12.8 GUI", "value": { "oeversion": "12.8", "graphicalMode": true, "buildPath": [ /* Whatever you want here */ ] }},
  { "name": "V12.8 TTY", "value": { "oeversion": "12.8", "graphicalMode": false }}
]
```

Profile can also inherit from another profile.

**Comment by M0rr155** [comment]
Ok that's clear. Thank you.

But what i am missing is how does the build action now when to compile with profile V12.8 GUI or with profile V12.8 TTY.

**Comment by M0rr155** [comment]
Is it possible to do something like this?

```
"buildPath": [
  // Entries can have type 'source' or 'propath'. Path attribute is mandatory. Build attribute is optional (defaults to 'path'). Xref attribute is optional (defaults to 'build/.pct' or '.builder/srcX')
  { "type": "source" , "path": "src/tty", "build": "build/tty", **"graphicalMode": false**},
```

**Comment by gquerret** [comment]
> Is it possible to do something like this?

No, the background session is either GUI or TTY, but it's not possible to specify the session mode per entry.
